### PR TITLE
test(gatsby-plugin-image): gatsby-image-server.ts

### PR DESCRIPTION
**Pull request by mfrachet**
## Description

Add basic tests for gatsby-image-server.ts. The following verifications have been made:

Screenshot 2020-11-12 at 13 22 20

Feel free to request more component level tests.

NB: I will also write Cypress tests when this one gets merged to verify older browser behaviours

### CODE REVIEW

File: packages/gatsby-plugin-image/src/components/gatsby-image.server.tsx [hunk 1]
```diff
@@ -57,6 +58,8 @@ export const GatsbyImage: FunctionComponent<GatsbyImageProps> = function GatsbyI
     }
   }
 
+  // QUESTION-1: sources is not used in getMainProps
+  // I'm not sure to understand the purpose.
```
Comment (mfrachet) [inline packages/gatsby-plugin-image/src/components/gatsby-image.server.tsx:62]: Summoning @ascorbic for this question :)

Comment (ascorbic) [reply packages/gatsby-plugin-image/src/components/gatsby-image.server.tsx:61]: It's assinged to cleanedImages below, which is passed to getMainProps, which then spreads it into the result. This means "sources" is then spread into MainImage

Comment (mfrachet) [reply packages/gatsby-plugin-image/src/components/gatsby-image.server.tsx:61]: Thank you, I've checked the code back and was isolating the test on img, while image is inside the picture tag. Sorry for bothering :/

**Comment by laurieontech** [comment]
This looks reasonable to me. It duplicates some of the static image stuff but I think that's fine.

**Review by gillkyle — APPROVED** [review]
These look great, I like the breakdown between styles, fallbacks, sources, and placeholders!

**Pull request merged**